### PR TITLE
[Misc] Add MKL library path for standalone installation

### PR DESCRIPTION
**Pull request by itaraban**
## Description

Fix sparse library link issue, caused by wrong Torch MKL CMake settings
Resolves issue: https://github.com/dmlc/dgl/issues/5474
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by daniil-sizov — APPROVED** [review]

File: dgl_sparse/CMakeLists.txt [hunk 1]
```diff
@@ -50,5 +50,12 @@ target_include_directories(${LIB_DGL_SPARSE_NAME} PRIVATE "${CMAKE_SOURCE_DIR}/t
 message(STATUS "DGL include directories: ${DGL_INCLUDE_DIRS}")
 target_include_directories(${LIB_DGL_SPARSE_NAME} PRIVATE ${DGL_INCLUDE_DIRS})
 target_link_directories(${LIB_DGL_SPARSE_NAME} PRIVATE ${DGL_BUILD_DIR} "${DGL_BUILD_DIR}/third_party/dmlc-core")
+
+# Torch CMake sets up MKL library path only for conda distribution
+# WA for standalone MKL installation
```
Comment (frozenbugs) [inline dgl_sparse/CMakeLists.txt:55]: what's WA?.

Comment (frozenbugs) [inline dgl_sparse/CMakeLists.txt:55]: nit, end the sentence with dot.

File: dgl_sparse/CMakeLists.txt [hunk 2]
```diff
@@ -50,5 +50,12 @@ target_include_directories(${LIB_DGL_SPARSE_NAME} PRIVATE "${CMAKE_SOURCE_DIR}/t
 message(STATUS "DGL include directories: ${DGL_INCLUDE_DIRS}")
 target_include_directories(${LIB_DGL_SPARSE_NAME} PRIVATE ${DGL_INCLUDE_DIRS})
 target_link_directories(${LIB_DGL_SPARSE_NAME} PRIVATE ${DGL_BUILD_DIR} "${DGL_BUILD_DIR}/third_party/dmlc-core")
+
+# Torch CMake sets up MKL library path only for conda distribution
+# Workaround for standalone MKL installation.
```
Comment (frozenbugs) [inline dgl_sparse/CMakeLists.txt:55]:
```suggestion
# The Torch CMake configuration only sets up the path for the MKL
# library when using the conda distribution. This is a workaround to
# address this when using a standalone installation of MKL.
```

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
FYI @czkkkkkk

File: dgl_sparse/CMakeLists.txt (hunk 2 above)
Comment (itaraban) [reply dgl_sparse/CMakeLists.txt]: thanks for suggestion!

**Comment by czkkkkkk** [comment]
Why does this PR fix #5474 ? In the issue, you mentioned that "libxsmm can use different compiler from what was used in cmake".  How does this PR fix the problem of libxsmm?

**Comment by itaraban** [comment]
Fix for "different compiler issue" was already in https://github.com/dmlc/dgl/pull/5497.
This PR fixes original problem - `cannot find -lmkl_core: No such file or directory`, which shouldn't require running `source /opt/intel/oneapi/setvars.sh`.

**Review by czkkkkkk — APPROVED** [review]

**Pull request merged**